sharktank/sharktank/export_layer/export_paged_attention.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Export support for the PagedLLMV1 protocol of models."""
+
+import json
+import torch
+
+from typing import Optional
+
+import torch.nn.functional as F
+
+from shark_turbine.aot import *
+
+from sharktank.layers import *
+from sharktank.types import *
+
+from sharktank.models.llama.testing import *
+from sharktank.layers import causal_llm
+
+# TODO: Should be using a base class with the protocol supported.
+from ..models.llama.llama import LlamaModelConfig, PagedLlamaAttentionBlock
+
+
+def paged_attention(
```

Comment:
Why is this function together with `run_llama` can not share the code with [paged_llama_attention_block.py](https://github.com/nod-ai/SHARK-Platform/blob/9f3f70f7f5018702abfa0dbb4954630179abb768/sharktank/sharktank/layers/paged_llama_attention_block.py)? They seem to be doing mostly the same things.

Reply:
This script is exporting a boiled down version of only the `sdpa` part of the paged attention found here -[paged_llama_attention_block.py](https://github.com/nod-ai/SHARK-Platform/blob/9f3f70f7f5018702abfa0dbb4954630179abb768/sharktank/sharktank/layers/paged_llama_attention_block.py).

Reply:
OK, so instead of doing flash attention, it is calling `torch.nn.functional.scaled_dot_product_attention`. Probably, in another PR we can have `sharktank.layers.PagedLlamaAttentionBlock` be configurable to accommodate for that scenario. This will avoid the code duplication.